conans/test/integration/build_environment_test.py

```diff
@@ -0,0 +1,232 @@
+import os
+import platform
+import unittest
+
+from conans.model.ref import ConanFileReference
+from conans.paths import CONANFILE
+from conans.test.tools import TestClient
+from conans.tools import unix_path
+from conans.util.files import md5sum
+
+mylibh = '''
+double mean(double a, double b);
+'''
+conanfile = '''
+from conans import ConanFile
+
+class ConanMeanLib(ConanFile):
+    name = "Mean"
+    version = "0.1"
+    exports_sources = "mean*"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"activate_define": [True, False], "optimize": [True, False]}
+    default_options = """activate_define=True
+optimize=False
+"""
+    generators = "gcc"
+
+    def build(self):
+        self.output.info("Running source!")
+        self.run("c++ -c -o mean.o @conanbuildinfo.gcc mean.cpp")
+        self.run("ar rcs libmean.a mean.o")
+
+    def package(self):
+        self.copy("*.h", dst="include")
+        self.copy("*.a", dst="lib")
+
+    def package_info(self):
+        self.cpp_info.libs.append("mean")
+        if self.options.activate_define:
+            self.cpp_info.defines.append("ACTIVE_VAR=1")
+        if self.options.optimize:
+            self.cpp_info.cflags.append("-O2")
+'''
+
+mylib = '''
+double mean(double a, double b) {
+  return (a+b) / 2;
+}
+'''
+
+example = '''
+#include "mean.h"
+#include <iostream>
+
+int main(){
+    std::cout << mean(10, 20) << std::endl;
+    #ifdef ACTIVE_VAR
+    std::cout << "Active var!!!" << std::endl;
+    #endif
+}
+    '''
+
+
+class BuildEnvironmenTest(unittest.TestCase):
+
+    def test_autotools_environment(self):
```

Comment:
remove, already tested with basic_build_test, diamond and some other with the cpp_test_files